api/src/main/java/com/nicknathanjustin/streamercontracts/contracts/dtos/PrivateContract.java

```diff
@@ -0,0 +1,23 @@
+package com.nicknathanjustin.streamercontracts.contracts.dtos;
+
+import com.nicknathanjustin.streamercontracts.contracts.ContractModel;
+import com.nicknathanjustin.streamercontracts.votes.VoteModel;
+
+import lombok.Data;
+import lombok.NonNull;
+
+@Data
+public class PrivateContract extends Contract {
+
+    private boolean userHasVoted;
+    
+    public PrivateContract(@NonNull final ContractModel contract) {
+        super(contract);
+        this.userHasVoted = false;
+        for (VoteModel vote : contract.getVotes()) {
```

Comment:
Lambdas/Streams are the cool new java8 way to do loops:
```
contract.getVotes().stream().filter(vote -> vote.getVoter().getTwitchUsername().equals(contract.getStreamer().getTwitchUsername())))
```
^^filters the list done to votes where the voter is the streamer. If the list is greater than 0 we know the streamer voted... Disclaimer: I didnt test to see if this code compiles or works correctly.